Apps/Src/FaultState.c

```diff
@@ -8,20 +8,20 @@
 #include "Minions.h"
 static bool fromThread = false; //whether fault was tripped from thread
 
+
 static void ArrayMotorKill(void) {
     BSP_GPIO_Write_Pin(CONTACTORS_PORT, ARRAY_CONTACTOR_PIN, OFF);
     BSP_GPIO_Write_Pin(CONTACTORS_PORT, MOTOR_CONTACTOR_PIN, OFF);
     while(1){;} //nonrecoverable
 }
 
 static void nonrecoverableFaultHandler(){
+    Minion_Error_t mErr;
     //turn additional lights on to indicate critical error
-    BSP_GPIO_Write_Pin(LIGHTS_PORT, LEFT_BLINK_PIN, OFF);
-    Display_SetLight(LEFT_BLINK, ON);
-    BSP_GPIO_Write_Pin(LIGHTS_PORT, RIGHT_BLINK_PIN, OFF);
-    Display_SetLight(RIGHT_BLINK, ON);
-    BSP_GPIO_Write_Pin(LIGHTS_PORT, BRAKELIGHT_PIN, OFF);
-    Display_SetLight(CTRL_FAULT,ON); //turn on fault light
+    //Display_SetLight(LEFT_BLINK, ON);
+    //Display_SetLight(RIGHT_BLINK, ON);
+    Minion_Write_Output(BRAKELIGHT, false, &mErr);
```

Comment:
should probably turn the brakelight on if we fault out, if possible. However, considering there's a mutex in the call stack, and we don't know if the OS is working properly, this should probably be a raw BSP_GPIO call to that pin.